File: services/client-service/src/main/java/zw/co/microbank/clientservice/service/ClientService.java
package zw.co.microbank.clientservice.service;

import org.keycloak.representations.AccessTokenResponse;
import zw.co.microbank.clientservice.dto.ClientResponse;
import zw.co.microbank.clientservice.dto.CreateClientRequest;
import zw.co.microbank.clientservice.dto.LoginRequest;

/**
 * Author: tjc
 * Created on 7/10/25
 */

public interface ClientService {
    // Client Creation and Registration
    ClientResponse createClient(CreateClientRequest request);

    // Client Login
    AccessTokenResponse loginClient(LoginRequest request);

    //    Fetch current client profile
    ClientResponse fetchClientProfile(String id);
    
    // Fetch client profile by Keycloak ID
    ClientResponse fetchClientProfileByKeycloakId(String keycloakId);
}

File: client/services/client.service.ts
import { apiGet, apiPost } from '@/lib/api'
import { ClientResponse, CreateClientRequest, LoginRequest } from '@/types'

export class ClientService {
  static async createClient(data: CreateClientRequest): Promise<ClientResponse> {
    return apiPost<ClientResponse>('/clients/register', data, false)
  }

  static async loginClient(data: LoginRequest): Promise<ClientResponse> {
    return apiPost<ClientResponse>('/clients/login', data, false)
  }

  static async getClientProfile(id: string): Promise<ClientResponse> {
    return apiGet<ClientResponse>(`/clients/profile/${id}`)
  }
}